lib/common.js

```diff
@@ -67,8 +129,9 @@ let requestOverride = []
  *   - overriddenRequest - the overridden module's request function already bound to module's object
  *   - options - the options of the issued request
  *   - callback - the callback of the issued request
+ * @param {boolean} legacy
  */
-const overrideRequests = function(newRequest) {
+const overrideRequests = function(newRequest, legacy) {
```

Comment:
Hi! so sorry to be just now getting back to this!

I see you've added a parameter here, which it seems like must be set to `true` when running in Node 8 and `false` otherwise.

Could you do this in code instead, so the caller doesn't need figure out the correct value for this parameter? We used to have code that did `semver.satisfies(process.version, '>=8')` and I think the same approach would work well here (with `'<10'`).

You can look at #1318 where one of these conditional checks was removed.